src/utils/order-messages.ts

```diff
@@ -74,20 +80,51 @@ export function checkAndParseOrderMessage(txt: string): SimpleOrder {
         }
     }
 
-    return { items: orderItems, remarks }
+    // Extract payment gateway details
+    const totalPaymentLines = lines.findIndex((line) => line.trim().startsWith(PAYMENT_GATEWAY_ID_LABEL))
+    if(totalPaymentLines !== -1){
+        const paymentGatewayLine = lines[totalPaymentLines+1]
+        if(paymentGatewayLine && paymentGatewayLine.trim() !== ''){
+            paymentGatewayId = paymentGatewayLine.trim()
+        }
+    }
+
+    return { items: orderItems, remarks, paymentGatewayId }
+}
+
+function serialisePreOrderMessage(shopName:string):string {
```

Comment:
these don't actually need to be separate fns -> can just be inside serialiseOrderMessage